janus_core/cli.py

```diff
@@ -362,19 +451,66 @@ def geomopt(  # pylint: disable=too-many-arguments,too-many-locals
     # Otherwise override with None
     fully_opt_dict = {} if (fully_opt or vectors_only) else {"filter_func": None}
 
-    # Run geometry optimization and save output structure
-    optimize(
-        s_point.struct,
-        fmax=fmax,
-        steps=steps,
-        filter_kwargs=filter_kwargs,
+    # Dictionary of inputs for optimize function
+    optimize_kwargs = {
+        "struct": s_point.struct,
+        "fmax": fmax,
+        "steps": steps,
+        "filter_kwargs": filter_kwargs,
         **fully_opt_dict,
-        opt_kwargs=opt_kwargs,
-        write_results=True,
-        write_kwargs=write_kwargs,
-        traj_kwargs=traj_kwargs,
-        log_kwargs={"filename": log_file, "filemode": "a"},
-    )
+        "opt_kwargs": opt_kwargs,
+        "write_results": True,
+        "write_kwargs": write_kwargs,
+        "traj_kwargs": traj_kwargs,
+        "log_kwargs": {"filename": log_file, "filemode": "a"},
+    }
+
+    # Store inputs for yaml summary
+    inputs = {}
+    for key, value in optimize_kwargs.items():
+        inputs[key] = value
+
+    # Store only filename as filemode is not set by user
+    del inputs["log_kwargs"]
+    inputs["log"] = log_file
+
+    inputs["struct"] = {
+        "n_atoms": len(s_point.struct),
+        "struct_path": struct_path,
+        "struct_name": s_point.struct_name,
+        "formula": s_point.struct.get_chemical_formula(),
+    }
+
+    inputs["calc"] = {
+        "architecture": architecture,
+        "device": device,
+        "read_kwargs": read_kwargs,
+        "calc_kwargs": calc_kwargs,
+    }
+
+    # Convert all paths to strings in inputs nested dictionary
+    _iter_path_to_str(inputs)
+
+    # Save summary information before optimization begins
+    save_info = [
+        {"command": "janus geomopt"},
+        {"start_time": datetime.datetime.now().strftime("%d/%m/%Y, %H:%M:%S")},
+        {"inputs": inputs},
+    ]
+    with open(summary, "w", encoding="utf8") as outfile:
+        yaml.dump(save_info, outfile, default_flow_style=False)
+
+    # Run geometry optimization and save output structure
+    optimize(**optimize_kwargs)
+
+    # Time after optimization has finished
+    with open(summary, "a", encoding="utf8") as outfile:
+        yaml.dump(
+            [{"end_time": datetime.datetime.now().strftime("%d/%m/%Y, %H:%M:%S")}],
+            outfile,
+            default_flow_style=False,
+        )
```

Comment:
Would it be better to pass `outfile` into the function and write directly rather than opening and closing it, or does that not work?

Reply:
I can't see an obvious way that we could pass the CLI command functions a file.